### DISCUSSION TITLE
Place YouTube links in a web part

### DISCUSSION THREAD

**Discussion by haroldvandekamp** [WikiTransformationProject/wikitraccs-releases#44] [Ideas]
Currently, links to YouTube are pasted as text links instead of being placed in a YouTube web part. Can this be integrated in the migration process?

So instead of:
![image](https://github.com/WikiTransformationProject/wikitraccs-releases/assets/11357196/eb30965a-7146-496c-8aa7-30d40f86abc1)
, place it in the web part like:
![image](https://github.com/WikiTransformationProject/wikitraccs-releases/assets/11357196/8cd4089e-f9b2-4a13-97ce-8b422de21ffc)

**Comment by heinrich-ulbricht** [comment]
@haroldvandekamp I'd love WikiTraccs to do that.

But there is one thing that makes me wonder if there is a useful way to do this in SharePoint.

In SharePoint web parts are rather heavy-weight. They cannot be embedded within other text, so no other web parts can be placed into the text web part itself. Confluence is different since you can place macros within the text anywhere. This is not possible in SharePoint.

Here's a sample Confluence page containing a YouTube video at two different places:

1. between two paragraphs
2. inside a table

![test2](https://github.com/WikiTransformationProject/wikitraccs-releases/assets/3469970/3af9a59b-45f6-48c3-b2b3-a75ae38726e8)

WikiTraccs could handle the first case well, without changing the page structure too much. One text web part, one YouTube web part, one text web part.

But how to handle the second case?

I looked at how SharePoint handles pasted YouTube links - it pastes the link as text into the text web part and adds an additional, new video web part _below_ the text web part.

@haroldvandekamp What would your expectation be when migrating pages with YouTube videos from Confluence to SharePoint?

**Comment by haroldvandekamp** [comment]
@heinrich-ulbricht : Thanks for the comprehensive explanation. That provides great context to discuss with our customer.

I now understand that for this scenario SharePoint is limited in functionality compared to Confluence. In this case, I would love that the migration tooling adds the YouTube web part _below_ the text web part, so similar to how SharePoint acts in this situation out of the box. Finalizing a page and manually restructuring the web parts on the page is easier than having to add those web parts manually to the page.

So, I think there are three options for how the migration tooling behaves:
1. Current situation where YouTube links are added as text links
2. Proposed situation where YouTube links are added as text links and YouTube web part is placed _under_ the text web part
3. YouTube link is not displayed, only YouTube web part is placed _under_ the text web part

For me, option 2 would be the best for now. 
Ideally, having a configuration setting where we can change the migration tooling behaviour of setting the preferred option would be the best of course :)!

**Comment by heinrich-ulbricht** [comment]
@haroldvandekamp This has been a very productive discussion :) I created a feature request (#45) and put it on the [roadmap](https://github.com/orgs/WikiTransformationProject/projects/3). Let's continue there if something comes up. This is a nice thing to have.
